### PR TITLE
multiple formatters

### LINKED ISSUE #90: Reporting - Output to file
It is currently not possible to specify an "output" file as a CLI argument for saving reports. The JSON formatter seems to write directly to STDOUT, which can be a problem when debugging (using console.log) or if the program fails, producing invalid JSON.

It would be desirable if this worked like cucumber in ruby, allowing multiple formats saved out to separate output files.

### LINKED ISSUE #91: Multiple formatters
Allow for multiple formatters in a single run. This is already easy to accomplish in non-CLI environments.

### PR DESCRIPTION
**Pull request by charlierudolph**
@jbpros @samccone 

fixes #91
fixes #90

### CODE REVIEW

File: lib/cucumber/cli/argument_parser.js [hunk 1]
```diff
@@ -106,9 +107,19 @@ function ArgumentParser(argv) {
...
+      var formats = self.getOptionOrDefault(ArgumentParser.FORMAT_OPTION_NAME, [ArgumentParser.DEFAULT_FORMAT_VALUE]);
+      var outputMapping = {};
+      formats.forEach(function (format) {
+        var parts = format.split(':');
+        var type = parts[0];
+        var outputTo = parts[1] || '';
+        outputMapping[outputTo] = type;
+      });
+      return _.map(outputMapping, function (type, outputTo) {
+        var stream = outputTo ? fs.createWriteStream(outputTo) : process.stdout;
```
Comment (jbpros) [inline lib/cucumber/cli/argument_parser.js]: Passing a bad path will currently silently fail. We should listen to errors (`.on('error', ...`) and fail the whole process if something wrong happens.

Comment (jbpros) [inline lib/cucumber/cli/argument_parser.js]: @charlierudolph could you add a test for that case as well? Cucumber's return code should also be non-zero.

Comment (charlierudolph) [inline lib/cucumber/cli/argument_parser.js]: @jbpros added

**Comment by jbpros** [comment]
Thanks for the update Charlie -- this is an exciting one!

:+1: LGTM!
